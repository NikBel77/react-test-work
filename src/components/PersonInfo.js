import React from 'react'

function PersonInfo(props) {

    if (props.personInfo) {
        if (!props.personInfo.description || !props.personInfo.address) {
            return (
                <div className='container'>
                    <p>no user address</p>
                </div>
            )
        }

        const name = `${props.personInfo.firstName} ${props.personInfo.lastName}`;
        const { city, state, streetAddress, zip } = props.personInfo.address;
        const description = props.personInfo.description;

        return (
            <div className='container'>
                <div className='selectedPerson'>
                    Selected user: <b>{name}</b><br />
                    Description:<br />
                    <textarea value={description} readOnly="readonly"></textarea>
                    <address>
                        City: <b>{city}</b><br />
                        State: <b>{state}</b><br />
                        Address: <b>{streetAddress}</b><br />
                        Zip: <b>{zip}</b><br />
                    </address>
                </div>
            </div>
        )
    } else {
        return (
            <div className='container'>
                <p>no user selected</p>
            </div>
        )
    }

}

export default PersonInfo