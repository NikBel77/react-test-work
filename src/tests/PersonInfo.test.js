import React from 'react'
import ReactDOM from 'react-dom'
import PersonInfo from '../components/PersonInfo'
import renderer from 'react-test-renderer'

const testPerson = {
    id: 101,
    firstName: 'Sue',
    lastName: 'Corson',
    email: 'DWhalley@in.gov',
    phone: '(612)211-6296',
    address: {
        streetAddress: '9792 Mattis Ct',
        city: 'Waukesha',
        state: 'WI',
        zip: '22178'
    },
    description: 'et lacus magna dolor...',
}
const shortPersonInfo = {
    id: 101,
    firstName: 'Sue',
    lastName: 'Corson',
    email: 'DWhalley@in.gov',
    phone: '(612)211-6296',
}


test('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<PersonInfo personInfo={testPerson} />, div);
    ReactDOM.unmountComponentAtNode(div);
});
test('render without crashing 2', () => {
    const div = document.createElement('div');
    ReactDOM.render(<PersonInfo personInfo={shortPersonInfo} />, div);
    ReactDOM.unmountComponentAtNode(div);
});
test('snapshot 1', () => {
    const tree = renderer.create(<PersonInfo personInfo={testPerson} />).toJSON();
    expect(tree).toMatchSnapshot();
});
test('snapshot 2', () => {
    const tree = renderer.create(<PersonInfo personInfo={shortPersonInfo} />).toJSON();
    expect(tree).toMatchSnapshot();
});