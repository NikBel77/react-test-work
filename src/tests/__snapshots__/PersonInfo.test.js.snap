// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1 1`] = `
<div
  className="container"
>
  <div
    className="selectedPerson"
  >
    Selected user: 
    <b>
      Sue Corson
    </b>
    <br />
    Description:
    <br />
    <textarea
      readOnly="readonly"
      value="et lacus magna dolor..."
    />
    <address>
      City: 
      <b>
        Waukesha
      </b>
      <br />
      State: 
      <b>
        WI
      </b>
      <br />
      Address: 
      <b>
        9792 Mattis Ct
      </b>
      <br />
      Zip: 
      <b>
        22178
      </b>
      <br />
    </address>
  </div>
</div>
`;

exports[`snapshot 2 1`] = `
<div
  className="container"
>
  <p>
    no user address
  </p>
</div>
`;
